api/handlers_e2e_test.go

```diff
@@ -300,6 +302,98 @@ func TestPagingRootTxnDeduplication(t *testing.T) {
 	}
 }
 
+func TestTransactionHandler(t *testing.T) {
+	db, shutdownFunc := setupIdb(t, test.MakeGenesis(), test.MakeGenesisBlock())
+	defer shutdownFunc()
+
+	///////////
+	// Given // A block containing a key reg txn with state proof key
+	///////////
+	var votePK crypto.OneTimeSignatureVerifier
+	votePK[0] = 1
+
+	var selectionPK crypto.VRFVerifier
+	selectionPK[0] = 1
+
+	var sprfkey merklekeystore.Verifier
+	sprfkey[0] = 1
+
+	txn := transactions.SignedTxnWithAD{
+		SignedTxn: transactions.SignedTxn{
+			Txn: transactions.Transaction{
+				Type: "keyreg",
+				Header: transactions.Header{
+					Sender:      test.AccountA,
+					GenesisHash: test.GenesisHash,
+				},
+				KeyregTxnFields: transactions.KeyregTxnFields{
+					VotePK: votePK,
+					SelectionPK:selectionPK,
+					StateProofPK: sprfkey,
+					VoteFirst: basics.Round(0),
+					VoteLast: basics.Round(100),
+					VoteKeyDilution: 1000,
+					Nonparticipation: false,
+				},
+			},
+			Sig: test.Signature,
+		},
+	}
+
+	block, err := test.MakeBlockForTxns(test.MakeGenesisBlock().BlockHeader, &txn)
+	require.NoError(t, err)
+
+	err = db.AddBlock(&block)
+	require.NoError(t, err, "failed to commit")
+
+	//////////
+	// When // We query the txn
+	//////////
+
+	e := echo.New()
```

Comment:
I would put this and next section in its own scope so that the section after doesn't accidentally use any of the old variables.
```
{
  req := httptest.NewRequest(http.MethodGet, "/", nil)
  ...
  require.Equal(t, json.Encode(sprfkey), json.Encode(*response.Transaction.KeyregTransaction.StateProofKey))
}
```